### PR TITLE
i2239: Use SRU instead of Alma API for bib record retrieval

### LINKED ISSUE #2239: Change `bibliographic` endpoint to use SRU rather than API
- When we have a crawler event that hits certain Orangelight endpoints (e.g. staff_view, .ris, .endnote), it can quickly use up our Alma api calls
- Currently, these calls go to the bibliographic endpoint (`"#{Requests.config['bibdata_base']}/bibliographic/#{id}"`, which then calls the Alma API

## Acceptance criteria
- The data provided from the bibliographic endpoint does not change
- The data is sourced from a call to the [Alma SRU](https://developers.exlibrisgroup.com/alma/integrations/sru/), rather than the Alma API.

### PR DESCRIPTION
**Pull request by sandbergja**
closes #2239

### CODE REVIEW

File: app/adapters/alma_adapter.rb [hunk 1]
```diff
@@ -11,28 +13,26 @@ def initialize(connection: AlmaAdapter::Connector.connection)
   # @param id [String] e.g. id = "991227830000541"
   # @see https://developers.exlibrisgroup.com/console/?url=/wp-content/uploads/alma/openapi/bibs.json#/Catalog/get%2Falmaws%2Fv1%2Fbibs Values that could be passed to the alma API
   # get one bib record is supported in the bibdata UI and in the bibliographic_controller
-  # @return [MARC::Record]
+  # @return [AlmaAdapter::MarcRecord]
```
Comment (sandbergja) [inline app/adapters/alma_adapter.rb:16]: Updating an outdated comment

File: app/adapters/alma_adapter.rb [hunk 2]
```diff
@@ -11,28 +13,26 @@ def initialize(connection: AlmaAdapter::Connector.connection)
...
   def get_bib_record(id, suppressed: true)
+    return nil unless /\A\d+\z/.match? id
     get_bib_records([id], suppressed:)&.first
   end
 
   # Get /almaws/v1/bibs Retrieve bibs
   # @param ids [Array] e.g. ids = ["991227850000541","991227840000541","99222441306421"]
   # @see https://developers.exlibrisgroup.com/console/?url=/wp-content/uploads/alma/openapi/bibs.json#/Catalog/get%2Falmaws%2Fv1%2Fbibs Values that could be passed to the alma API
-  # @return [Array<MARC::Record>]
+  # @return [Array<AlmaAdapter::MarcRecord>]
```
Comment (sandbergja) [inline app/adapters/alma_adapter.rb:25]: Updating outdated comment

File: app/adapters/alma_adapter.rb [hunk 3]
```diff
@@ -11,28 +13,26 @@ def initialize(connection: AlmaAdapter::Connector.connection)
   # @param id [String] e.g. id = "991227830000541"
   # @see https://developers.exlibrisgroup.com/console/?url=/wp-content/uploads/alma/openapi/bibs.json#/Catalog/get%2Falmaws%2Fv1%2Fbibs Values that could be passed to the alma API
   # get one bib record is supported in the bibdata UI and in the bibliographic_controller
-  # @return [MARC::Record]
-  def get_bib_record(id, suppressed: true)
-    get_bib_records([id], suppressed:)&.first
+  # @return [AlmaAdapter::MarcRecord]
+  def get_bib_record(id, show_suppressed: false)
```
Comment (sandbergja) [inline app/adapters/alma_adapter.rb:17]: I found the `suppressed` argument (which displayed suppressed records when false, and did not display them when true) confusing.  I switched the behavior around and tried to clarify the name.

**Review by maxkadel — APPROVED** [review]
Thanks @sandbergja! I appreciate the test helper method! 

Checked catalog-staging (pointing to bibdata-staging), including staff_view, .endnote, and .ris formats, and everything looks good!

**Pull request merged**